src/lib.rs

```diff
@@ -230,6 +230,39 @@ impl<'b, SC: max_stack::Config<'b>, GC: gas::Config<'b>> Analysis<SC, GC> {
     }
 }
 
+/// The fee to charge at the instrumentation point.
+///
+/// Each fee follows the formula of `a + bx` where the `b * x` component allows to adjust the
+/// fee based on a single input argument. The specific interpretation of the input arguments
+/// depends on how the instrumentation is implemented. The instrumentation provided by the
+/// `instrument` module, for example, introduces separate imports of gas instrumentation functions
+/// for each aggregate instruction supported by this module.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct Fee {
+    /// Constant fee factor.
+    pub constant: u64,
+    /// Linear fee factor.
+    pub linear: u64,
+}
+
+impl Fee {
+    const ZERO: Fee = Fee {
+        constant: 0,
+        linear: 0,
+    };
+
+    pub fn constant(constant: u64) -> Self {
+        Self { constant, linear: 0 }
+    }
+
+    pub(crate) fn checked_add(self, other: Fee) -> Option<Self> {
+        Some(Self {
+            linear: (self.linear == 0 && other.linear == 0).then_some(other.linear)?,
+            constant: self.constant.checked_add(other.constant)?,
+        })
+    }
```

Comment:
Double-checked this and this is fine. Whether instrumentation point can be merged or not is primarily determined by the `InstrumentationKind`. In past code whether gas can be added was only important for handling of possible overflow.

In principle this could be relaxed somewhat to allow for the 1st instrumentation point with linear fee to merge in any further constant fees into itself, but that makes this function not commutative, and I don't want to review all of the implications of that at this point in time, so we won't allow merging in any case where fees are non-constant for now.